File: medicare-backend/app/models/subscription.py
"""Subscription ORM model."""
import uuid

from sqlalchemy import ForeignKey, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin, UUIDMixin


class Subscription(Base, UUIDMixin, TimestampMixin):
    __tablename__ = "subscriptions"

    customer_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("customers.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    vertical: Mapped[str] = mapped_column(String(100), nullable=False)
    plan: Mapped[str] = mapped_column(String(100), nullable=False, default="basic")
    status: Mapped[str] = mapped_column(
        String(50),
        nullable=False,
        default="active",
        # CHECK: active | paused | cancelled | expired
    )
    renewal_date: Mapped[str | None] = mapped_column(String(20), nullable=True)

File: medicare-frontend/src/types/subscription.ts
export type SubscriptionStatus =
  | "active"
  | "paused"
  | "cancelled"
  | "expired";

export interface Subscription {
  id: string;
  customer_id: string;
  product_id: string;
  variant_id: string;
  status: SubscriptionStatus;
  next_billing_date: string | null;
  created_at: string;
}
